### PR TITLE
ISPN-14606 ZUNION, ZINTER, ZUNIONSTORE, ZINTERSTORE

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-14606

### CODE REVIEW

**Review by jabolina — COMMENTED** [review]
Just one small suggestion, everything looks good, though.

Also, does Redis has some sort of isolation for the sets accessed in this operation? I couldn't find anything. If so, maybe we add a note in the documentation.

**Comment by karesti** [comment]
> Just one small suggestion, everything looks good, though.
> 
> Also, does Redis has some sort of isolation for the sets accessed in this operation? I couldn't find anything. If so, maybe we add a note in the documentation.

As you said, in the docs nothing is specified...

File: server/resp/src/main/java/org/infinispan/server/resp/commands/sortedset/internal/AGGCommand.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+         }
+      }
+
+      EmbeddedMultimapSortedSetCache<byte[], byte[]> sortedSetCache = handler.getSortedSeMultimap();
+      final SortedSetBucket.AggregateFunction finalAggFunction = aggOption;
+      CompletionStage<Collection<SortedSetBucket.ScoredValue<byte[]>>> aggValues;
+      if (aggCommandType == AGGCommandType.UNION) {
+         aggValues = sortedSetCache.union(keys.get(0), null, computeWeight(weights, 0), finalAggFunction);
+      } else {
+         aggValues = sortedSetCache.inter(keys.get(0), null, computeWeight(weights, 0), finalAggFunction);
```
[the remark this answers is not in the document]
Comment (karesti) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/sortedset/internal/AGGCommand.java:129]: yes, we need to grab the values and apply the weight for each of the, from the first sorted set.

**Comment by karesti** [comment]
@jabolina I added your change

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Thanks, @karesti!

**Pull request merged**